apps/meteor/app/api/server/v1/channels.ts

```diff
@@ -471,8 +471,7 @@ API.v1.addRoute(
 
 			const room = findChannelByIdOrName({
 				params: {
-					roomId: channelId,
-					roomName: channelName,
+					...(channelId ? { roomId: channelId } : { roomName: channelName }),
```

Comment:
Do we have to do it like this?

**In this particular context**, `roomId in params` is the same as `params.roomId ?` - so no matter what you do in the calling function, the called **still** has to check for both values any way.

This conditional and destructuring just seems useless at that point. And IMO is bad. We should keep the code simple. And inspire others (people who are gonna read your code) to do so too.

on the called, you can simply change the param type from the union to a simple all-partial and chain the truthness checks `roomId ? .. : roomName ? ... : undefined`

it's late for me, hopefully i was able to explain myself.

Reply:
Hey Debdut, I'll first explain what was the issue on the way we were _calling_ this function (that is, with `params: { roomId: channelId, roomName: channelName }`): this would cause one of the fields in the `params` object to be `undefined`. So checking `'roomId' in params` in `findChannelByIdOrName` would _always_ return `true` (since this field is in the object, even though its value is `undefined`).

Still, I see we could change the params type to `params: { roomId?: string; roomName?: string }` and check for `params.roomId` instead of `'roomId' in params` (is that what you mean?). If we do so, I think we should also bring back the error check we had before (so as to throw an error when both params are undefined).

Reply:
> throw an error when both params are undefined

Already done by the route function. So, you can safely ignore that part.

Reply:
https://github.com/RocketChat/Rocket.Chat/blob/aaebe90bb2a1bbd72c0c2b0c33bd38849b9058ef/apps/meteor/app/api/server/v1/channels.ts#L464-L466